src/Table.js

```diff
@@ -0,0 +1,189 @@
+            <div className="product-price">{props.productPrice}</div>
+        </div>
+    );
+}
+
+function ProductCategoryRow(props) {
+    return (
+        <div className="category-row">
+            {props.category}
+        </div>
+    );
+}
+
+function ProductTable(props) {
+    const filterText = props.filterText;
+    const inStockOnly = props.inStockOnly;
+    const table = [...data];
+    const filteredTable = filterTable(table, filterText, inStockOnly);
+    //filteredTable.map((v) => console.log("ProductTable.name: " + v.name));
+    //console.log("Size:  " + filteredTable.length);  
+    return (
+        <div className="table">
+            <div className="row">
+                <div className="product-desc">Name</div>
+                <div className="product-desc">Price</div>
+            </div>
+            <div>
+                {filteredTable.map((v, i, tab) => {
+                    if(i === 0) {
+                        return (
+                            <Fragment key={v.name} >
+                                <ProductCategoryRow category={v.category} />
+                                <ProductRow productName={v.name} productPrice={v.price} productStocked={v.stocked} />
+                            </Fragment>
+                        );
+                    } else if (i > 0 && (v.category !== tab[i - 1].category)) {
+                        return (
+                            <Fragment key={v.name} >
+                                <ProductCategoryRow category={v.category} />
+                                <ProductRow productName={v.name} productPrice={v.price} productStocked={v.stocked} />
+                            </Fragment>
+                        );
+                    } else {
+                        return (
+                            <Fragment key={v.name} >
+                                <ProductRow productName={v.name} productPrice={v.price} productStocked={v.stocked} />
+                            </Fragment>
+                        );
+                    }
+                })}
+            </div>
+        </div>
+    );
+}
+
+function SearchBar(props) {
+    const filterText = props.filterText;
+    const inStockOnly = props.inStockOnly;
+    //console.log("SearchBar: " + inStockOnly + ",  " + filterText);
+    return (
+        <div className="search">
+            <div className="search-bar">
+                <input
+                    id="search-bar-input" 
+                    className="search-bar-input" 
+                    type="text" 
+                    placeholder="Search..." 
+                    value={filterText}
+                    onChange={props.onChangeFilterText}
+                />
+            </div>
+            <div className="search-box">
+                <input 
+                    type="checkbox"
+                    id="in-stock-only"
+                    checked={inStockOnly}
+                    onChange={props.onChangeInStockOnly}
```

Comment:
skoro filterTex i inStockOnly przypisujesz do zmiennych to dla spójności warto było by również zrobić to samo z onChangeFilterText i onChangeInStockOnly